lib/src/transformers/take_last.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:async';
+
+import 'package:rxdart/src/utils/forwarding_sink.dart';
+import 'package:rxdart/src/utils/forwarding_stream.dart';
+
+class _TakeLastStreamSink<T> implements ForwardingSink<T, T> {
+  _TakeLastStreamSink(this.count);
+
+  final int count;
+  List<T> buffer = [];
+
+  @override
+  void add(EventSink<T> sink, T data) {
+    if (count > 0) {
+      buffer.add(data);
+
+      if (buffer.length > count) {
+        buffer.removeAt(0);
+      }
+    }
+  }
+
+  @override
+  void addError(EventSink<T> sink, dynamic e, [st]) => sink.addError(e, st);
+
+  @override
+  void close(EventSink<T> sink) {
+    buffer.forEach(sink.add);
+    buffer = [];
+    sink.close();
+  }
+
+  @override
+  FutureOr onCancel(EventSink<T> sink) {}
```

Comment:
Should clear list/queue on cancel.

Reply:
`onCancel` can be called multiple times, right? Could be a problem if the queue is cleared?

Reply:
When cancelling subscription or calling sink.close(), onCancel will be called.
For broadcast stream, onCancel can be called multiples times, but we only focus from `onListen` untill `onCancel`.

For example, I copied from `sdk`:
```dart
   * The returned stream is a broadcast stream if this stream is.
   * For a broadcast stream, the events are only counted from the time
   * the returned stream is listened to.
   */
  Stream<T> skip(int count) {
    return new _SkipStream<T>(this, count);
  }
```

```suggestion

  @override
  void close(EventSink<T> sink) {
    queue.forEach(sink.add);
    sink.close(); // <- also call onCancel
  }

  @override
  FutureOr onCancel(EventSink<T> sink) => queue.clear();

```

Reply:
OK that's perfect, thanks!